File: app/src/components/LoadingSpinner.js
import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';
import { colors } from '../constants/colors';

export default function LoadingSpinner({ size = 'large', fullScreen = true }) {
  if (!fullScreen) {
    return <ActivityIndicator size={size} color={colors.neonCyan} />;
  }
  return (
    <View style={styles.container}>
      <ActivityIndicator size={size} color={colors.neonCyan} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.background,
  },
});

File: Calcutta Node App/flutter_app/lib/widgets/loading_spinner.dart
import 'package:flutter/material.dart';
import '../constants/colors.dart';

class LoadingSpinner extends StatelessWidget {
  final bool fullScreen;
  const LoadingSpinner({super.key, this.fullScreen = true});

  @override
  Widget build(BuildContext context) {
    if (!fullScreen) {
      return const Center(child: CircularProgressIndicator(color: AppColors.neonCyan));
    }
    return const Scaffold(
      backgroundColor: AppColors.background,
      body: Center(child: CircularProgressIndicator(color: AppColors.neonCyan)),
    );
  }
}
